eth2util/deposit/deposit.go

```diff
@@ -188,8 +191,8 @@ func withdrawalCredsFromAddr(addr string) ([32]byte, error) {
 	}
 
 	var creds [32]byte
-	copy(creds[0:], eth1AddressWithdrawalPrefix) // Add 1 byte prefix.
-	copy(creds[12:], addrBytes)                  // Add 20 bytes of ethereum address suffix.
+	copy(creds[0:], eip7251AddressWithdrawalPrefix) // Add 1 byte prefix.
```

Comment:
Will validator manually migrate to 0x02 credentials? If so, migration from 0x01 to 0x02 might take some time, meaning credentials for most of validators in the beginning will stay at `0x01...`. Have you double checked if this change won't break some stuff?

Also probably good to change the function description on top from `0x01` to `0x02`.

Reply:
AFAIK this code only applies to new cluster creations. It should be fine running a mix of 0x01 and 0x02 validators. For the old ones the users will not be able to use new features like partial withdrawals until the migration is complete.

Reply:
> Also probably good to change the function description on top from 0x01 to 0x02.

updated.